src/public-modules/Bounty/sagas.js

```diff
@@ -260,6 +292,207 @@ export function* getDraft(action) {
+  } catch (e) {
+    yield put(stdBountyFail());
+    yield put(setTransactionError());
+  }
+}
+
+export function* activateBounty(action) {
+  const { id, balance, paysTokens, decimals, tokenContract } = action;
+  const userAddress = yield select(addressSelector);
+  yield put(setPendingWalletConfirm());
+  let contractBalance;
+  if (paysTokens) {
+    contractBalance = calculateDecimals(
+      BigNumber(balance, 10).toString(),
+      parseInt(decimals)
+    );
+  } else {
+    const { web3 } = yield call(getWeb3Client);
+    contractBalance = web3.utils.toWei(
+      BigNumber(balance, 10).toString(),
+      'ether'
+    );
+  }
+  try {
+    const { standardBounties } = yield call(getContractClient);
+    let txHash;
+    if (paysTokens) {
+      const { tokenContract: tokenContractClient } = yield call(
+        getTokenClient,
+        tokenContract
+      );
+      const network = yield select(networkSelector);
+      const [approveHash, payoutHash] = yield call(
+        batchContractMethods,
+        [
+          tokenContractClient.approve(
+            config[network].standardBountiesAddress,
+            parseInt(contractBalance)
+          ).send,
+          { from: userAddress }
+        ],
+        [
+          standardBounties.activateBounty(id, parseInt(contractBalance)).send,
+          { from: userAddress }
+        ]
+      );
+      txHash = payoutHash;
+    } else {
+      txHash = yield call(
+        promisifyContractCall(standardBounties.activateBounty, {
+          from: userAddress,
+          value: contractBalance
+        }),
+        id,
+        contractBalance
+      );
+    }
+    yield put(stdBountySuccess());
+    yield put(setPendingReceipt(txHash));
+  } catch (e) {
+    yield put(stdBountyFail());
+    yield put(setTransactionError());
+  }
+}
+
+export function* extendDeadline(action) {
+  const { id, deadline } = action;
+  const userAddress = yield select(addressSelector);
+  yield put(setPendingWalletConfirm());
+  const formattedDeadline = parseInt(
+    moment(deadline)
+      .utc()
+      .toDate()
+      .getTime() / 1000
+  );
+
+  try {
+    const { standardBounties } = yield call(getContractClient);
+    const txHash = yield call(
+      promisifyContractCall(standardBounties.extendDeadline, {
+        from: userAddress
+      }),
+      id,
+      `${formattedDeadline}`
+    );
+    yield put(stdBountySuccess());
+    yield put(setPendingReceipt(txHash));
+  } catch (e) {
+    yield put(stdBountyFail());
+    yield put(setTransactionError());
+  }
+}
+
+export function* increasePayout(action) {
+  const {
+    id,
+    fulfillmentAmount,
+    balance,
+    paysTokens,
+    decimals,
+    tokenContract
+  } = action;
+  const userAddress = yield select(addressSelector);
+  yield put(setPendingWalletConfirm());
+
+  let contractFulfillmentAmount;
+  let contractBalance;
+  if (paysTokens) {
+    contractFulfillmentAmount = calculateDecimals(
+      BigNumber(fulfillmentAmount, 10).toString(),
```

Comment:
Would we ever want more or less accuracy than the `10` we're using here? Seeing it used several places, I wonder if it would make sense as a constant.